packages/edit-site/src/components/global-styles/screen-revisions/index.js

```diff
@@ -0,0 +1,155 @@
+	const { blocks, editorCanvasContainerView } = useSelect( ( select ) => {
+		return {
+			editorCanvasContainerView: unlock(
+				select( editSiteStore )
+			).getEditorCanvasContainerView(),
+			blocks: select( blockEditorStore ).getBlocks(),
+		};
+	}, [] );
+
+	const { revisions, isLoading, hasUnsavedChanges } =
+		useGetGlobalStylesRevisions();
+	const [ globalStylesRevision, setGlobalStylesRevision ] = useState( {} );
+
+	const [ currentRevisionId, setCurrentRevisionId ] = useState(
+		revisions?.[ 0 ]?.id
+	);
+	const [
+		isLoadingRevisionWithUnsavedChanges,
+		setIsLoadingRevisionWithUnsavedChanges,
+	] = useState( false );
+	const { setEditorCanvasContainerView } = unlock(
+		useDispatch( editSiteStore )
+	);
+
+	useEffect( () => {
+		if ( editorCanvasContainerView !== 'global-styles-revisions' ) {
+			goBack();
+			setEditorCanvasContainerView( editorCanvasContainerView );
+		}
+	}, [ editorCanvasContainerView ] );
+
+	const onCloseRevisions = () => {
+		goBack();
+	};
+
+	const restoreRevision = ( revision ) => {
+		setUserConfig( () => ( {
+			styles: revision?.styles,
+			settings: revision?.settings,
+		} ) );
+		setIsLoadingRevisionWithUnsavedChanges( false );
+		onCloseRevisions();
+	};
+
+	const selectRevision = ( revision ) => {
+		setGlobalStylesRevision( {
+			styles: revision?.styles,
+			settings: revision?.settings,
+			id: revision?.id,
+		} );
+		setCurrentRevisionId( revision?.id );
+	};
+
+	const isLoadButtonEnabled =
+		!! globalStylesRevision?.id &&
+		! isGlobalStyleConfigEqual( globalStylesRevision, userConfig );
+
+	return (
+		<>
+			<ScreenHeader
+				title={ __( 'Revisions' ) }
+				description={ __(
+					'Revisions are added to the timeline when style changes are saved.'
+				) }
+			/>
+			{ isLoading && (
+				<Spinner className="edit-site-global-styles-screen-revisions__loading" />
+			) }
+			{ ! isLoading && (
+				<Revisions
+					blocks={ blocks }
+					userConfig={ globalStylesRevision }
+					onClose={ onCloseRevisions }
+				/>
+			) }
+			<div className="edit-site-global-styles-screen-revisions">
+				<RevisionsButtons
+					onChange={ selectRevision }
+					currentRevisionId={ currentRevisionId }
+					userRevisions={ revisions }
+				/>
+				{ isLoadButtonEnabled && (
+					<SidebarFixedBottom>
+						<Button
+							variant="primary"
+							className="edit-site-global-styles-screen-revisions__button"
+							aria-label={ __( 'Load revision' ) }
+							disabled={
+								! globalStylesRevision?.id ||
+								globalStylesRevision?.id === 'unsaved'
+							}
+							onClick={ () => {
+								if ( hasUnsavedChanges ) {
+									setIsLoadingRevisionWithUnsavedChanges(
+										true
+									);
+								} else {
+									restoreRevision( globalStylesRevision );
+								}
+							} }
+						>
+							{ __( 'Load revision' ) }
+						</Button>
+					</SidebarFixedBottom>
+				) }
+			</div>
+			{ isLoadingRevisionWithUnsavedChanges && (
+				<RestoreGlobalStylesRevisionModal
```

Comment:
We should use the existing `ConfirmDialog` component instead of our own custom modal.

https://wordpress.github.io/gutenberg/?path=/story/components-experimental-confirmdialog--default

Reply:
**TIL** Much easier, thanks!